.github/workflows/build-kafka-images.yml

```diff
@@ -11,57 +11,126 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - amd64
+          - arm64
+    permissions:
+      contents: read
+      packages: write
+    name: Build Docker Images for ${{ matrix.arch }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```

Comment:
Aside: dependabot has the ability to keep these things current. https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot